### DISCUSSION TITLE
Query builder inline accordion component

### DISCUSSION THREAD

**Discussion by thunderpigeon** [grafana/grafana#38271] [Dashboards, Panels & Visualizations]
Hey @zoltanbedi i made a ticket for the component. As another project also needs it and therefore needs to know when it ready :) 

[Figma](https://www.figma.com/file/mogrxFlAlF4GL4xj87kKr6/Query-builder-and-explore-query-builder-jaeger-tempo?node-id=0%3A1)

**Where is/could it be used?**:

- [ ] query builders
- [ ] Jeager
- [ ] Complex queries

https://user-images.githubusercontent.com/21049979/116976224-271fe780-acc1-11eb-90d3-9e703ca2af1c.mov

___
- [ ] Post screenshots possible.
- [ ] It has a single use case.
- [ ] It is/could be used in multiple places.

**Implementation** (Checklist meant for the person implementing the component)

- [ ] Component has a story in Storybook.
- [ ] Props and naming follows [our style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/frontend.md).
- [ ] It is extendable (rest props are spread, styles with className work, and so on).
- [ ] Uses [theme for spacing, colors, and so on](https://github.com/grafana/grafana/blob/master/contribute/style-guides/themes.md).
- [ ] Works with both light and dark theme.

**Documentation**

- [ ] Properties are documented.
- [ ] Use cases are described.
- [ ] Code examples for the different use cases.
- [ ] Dos and don'ts.
- [ ] Styling guidelines, specific color usage (if applicable).

**Comment by clord** [comment]
To be consistent with other places where we do disclosures, I propose something that looks more like this:

<img width="606" alt="Screen Shot 2021-06-08 at 4 31 45 PM" src="https://user-images.githubusercontent.com/231203/121266005-04ac5b80-c877-11eb-81d7-617dc78011b3.png">

I've implemented this look for ML forms, and it works well. I needed a custom collapse so that form elements did not get removed from the DOM. I am willing to port it over to grafana/ui and close this issue with it.

**Comment by clord** [comment]
Here is an example of a disclosure in this style elsewhere:

<img width="248" alt="Screen Shot 2021-06-08 at 4 33 35 PM" src="https://user-images.githubusercontent.com/231203/121266178-489f6080-c877-11eb-903a-50e0d0003859.png">

**Comment by zoltanbedi** [comment]
From my side what you did looks good but let's hear @grafana/grafana-frontend-platform team if they want this component or not.

**Comment by tskarhed** [comment]
I would agree with @clord here. Not sure about the reason for the blue color of the text though.

**Comment by clord** [comment]
imo the blue calls attention to the interactive nature, similar to other places that have a similar mechanism. I originally had it white/gray but noticed many places with this pattern are blue. I can go back to white. I notice there is already `packages/grafana-ui/src/components/Collapse/Collapse.tsx` so could just update that with the delta.

**Comment by tskarhed** [comment]
Regardless, I see value in having the arrow to the left (straw accessibility test). The collapse has the same functionality, but in an entirely different context. Maybe create an inline version of it as a separate component? Any UX thoughts on these approaches? @Ijin08

**Comment by tskarhed** [comment]
The implementation is going to be put on hold, until the UX team has finished up the new inline forms.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
